src/CalculatorUITestFramework/MemoryPanel.cs

```diff
@@ -38,5 +51,112 @@ public ReadOnlyCollection<AppiumWebElement> GetAllMemoryListViewItems()
             OpenMemoryPanel();
             return this.MemoryListView.FindElementsByClassName("ListViewItem");
         }
+
+        /// <summary>
+        /// Opens the Memory Panel and clicks the delete button if it is visible
+        /// </summary>
+        public void ClearMemoryPanel()
+        {
+            this.MemoryLabel.Click();
+
+            try
+            {
+                string sourceC = this.session.PageSource;
+                if (sourceC.Contains("ClearMemoryButton"))
+                {
+                    this.ClearMemory.Click();
+                }
+                else
+                {
+                    return;
+                }
+            }
+            catch (WebDriverException ex)
+            {
+                if (ex.Message.Contains("element could not be located"))
+                {
+                    Assert.IsNotNull(this.MemoryPaneEmptyLabel);
+                    return;
+                }
+
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// If the Memory label is not displayed, resize the window
+        /// Two attempts are made, the the lable is not found a "not found" exception is thrown
+        /// </summary>
+        public void ResizeWindowToDiplayMemoryLabel()
+        {
+            Point newWindowPostion = new Point(0, 0);
+            WinAppDriver.Instance.CalculatorSession.Manage().Window.Position = newWindowPostion;
+            string source3 = this.session.PageSource;
+            if (source3.Contains("MemoryLabel"))
+            {
+                return;
+            }
+            else
+            {
+                Size newWindowSize = new Size(1200, 1050);
+                WinAppDriver.Instance.CalculatorSession.Manage().Window.Size = newWindowSize;
+                string source4 = this.session.PageSource;
+                if (source4.Contains("MemoryLabel"))
+                {
+                    return;
+                }
+                else
+                {
+                    Size newWindowSize2 = new Size(2097, 1282);
+                    WinAppDriver.Instance.CalculatorSession.Manage().Window.Size = newWindowSize2;
+                }
+                string source7 = this.session.PageSource;
+                if (source7.Contains("MemoryLabel"))
+                {
+                    return;
+                }
+                else
+                {
+                    throw new NotFoundException("Could not the Memory Label");
+                }
+            }
+        }
+
+        /// <summary>
+        /// If the Memory button is not displayed, resize the window
+        /// </summary>
+        public void ResizeWindowToDiplayMemoryButton()
+        {
+            Point newWindowPostion = new Point(0, 0);
+            WinAppDriver.Instance.CalculatorSession.Manage().Window.Position = newWindowPostion;
+            string source5 = this.session.PageSource;
+            if (source5.Contains("Alt, M"))
```

Comment:
Why are we looking for the hotkeys?

Reply:
Because of the way this check works.  Using memory button name gives a false positive.  Looking for this does not